doc/ref/states/requisites.rst

```diff
@@ -248,62 +195,117 @@ including the sls file and then setting a state to ``require`` the included sls
-dependent state. The state containing the ``require_any`` requisite is defined as the
-dependent state. The states specified in the ``require_any`` statement are defined as the
-required states. If at least one of the required state's execution succeeds, the dependent state
-will then execute.  If all of the executions by the required states fail, the dependent state
-will not execute.
+The ``onchanges`` requisite makes a state only apply if the required states
+generate changes, and if the watched state's "result" is ``True`` (does not fail).
+This can be a useful way to execute a post hook after changing aspects of a system.
+
+If a state has multiple ``onchanges`` requisites then the state will trigger
+if any of the watched states changes.
 
 .. code-block:: yaml
 
-    A:
-      cmd.run:
-        - name: echo A
-        - require_any:
-          - cmd: B
-          - cmd: C
-          - cmd: D
-    B:
+    myservice:
+      file.managed:
+        - name: /etc/myservice/myservice.conf
+        - source: salt://myservice/files/myservice.conf
       cmd.run:
-        - name: echo B
+        - name: /usr/local/sbin/run-build
+        - onchanges:
+          - file: /etc/myservice/myservice.conf
 
-    C:
-      cmd.run:
-        - name: /bin/false
+In the example above, ``cmd.run`` will run only if there are changes in the
+``file.managed`` state.
 
-    D:
+An easy mistake to make is using ``onchanges_in`` when ``onchanges`` is the
+correct choice, as seen in this next example.
+
+.. code-block:: yaml
+
+    myservice:
+      file.managed:
+        - name: /etc/myservice/myservice.conf
+        - source: salt://myservice/files/myservice.conf
       cmd.run:
-        - name: echo D
+        - name: /usr/local/sbin/run-build
+        - onchanges_in:  # <-- broken logic
+          - file: /etc/myservice/myservice.conf
 
-In this example `A` will run because at least one of the requirements specified,
-`B`, `C`, or `D` will succeed.
+
+This will set up a requisite relationship in which the ``cmd.run`` state
+always executes, and the ``file.managed`` state only executes if the
+``cmd.run`` state has changes (which it always will, since the ``cmd.run``
+state includes the command results as changes).
+
+It may semantically seem like the ``cmd.run`` state should only run
+when there are changes in the file state, but remember that requisite
+relationships involve one state watching another state, and a
+:ref:`requisite_in <requisites-onchanges-in>` does the opposite: it forces
+the specified state to watch the state with the ``requisite_in``.
 
 .. _requisites-watch:
 
 watch
 ~~~~~
 
-``watch`` statements are used to add additional behavior when there are changes
-in other states.
+A ``watch`` requisite is used to add additional behavior when there are changes
+in other states. This is done using the ``mod_watch`` function available from
+the execution module and will execute any time a ``watch``\ed state changes.
```

Comment:
Is this escape char intentional?

Reply:
In my viewer, it was required to render \`\`watch\`\`ed as ``watch``ed. Apparently that's a local limitation. I'll remove it.